src/todos/todos.guard.ts

```diff
@@ -0,0 +1,30 @@
+import { TodosService } from './todos.service';
+import {
+  Injectable,
+  CanActivate,
+  ExecutionContext,
+  Inject,
+  forwardRef,
+} from '@nestjs/common';
+import { Observable } from 'rxjs';
+
+@Injectable()
+export class TodosGuard implements CanActivate {
```

Comment:
Guard này thường là check authen, role trước khi vào Resolved anh nha, thường dùng chung cho toàn bộ dự án, nên cố gắng viết chung, hạn chế theo từng cái anh nhé